### ISSUE TITLE
[Data Mapper] Render links for query expressions resides within member access expressions

### ISSUE THREAD

**Issue by madushajg** [wso2/ballerina-plugin-vscode#366]
**Description:**
Currently in the Data Mapper, the recursive query editing is only possible if both sides of the mapping are arrays. But users can use query expressions, access the returned members and use in the target type.
ie:
```ballerina
function transform(Person person) returns Student => {
    contactPhone: (from var contact in registry
        where contact.mediumType == "TELEPHONE"
        select contact.phoneNumber)[0]
}
```

For these type of mappings also, the DM should render a query expression node similar to array-to-array mappings with query expressions.

**Comment by janihur** [comment]
Does this also cover cases where the same top level array is used to derive values for several different top level values. This kind of structure is very common in my current work:

From:
```
  "relatedParty": [
    {
      "name": "RELATEDPARTY_1_NAME",
      "role": "ORDER_CONTACT",
      "contactMedium": [
        {
          "mediumType": "EMAIL",
          "characteristic": {
            "emailAddress": "RELATEDPARTY_1_EMAILADDRESS"
          }
        },
        {
          "mediumType": "TELEPHONE",
          "characteristic": {
            "phoneNumber": "RELATEDPARTY_1_PHONENUMBER"
          }
        },
        {
          "mediumType": "POSTAL_ADDRESS",
          "characteristic": {
            "city": "RELATEDPARTY_1_CITY",
            "country": "RELATEDPARTY_1_COUNTRY",
            "postCode": "RELATEDPARTY_1_POSTCODE",
            "street1": "RELATEDPARTY_1_STREET1",
            "name": "RELATEDPARTY_1_NAME"
          }
        }
      ]
    }
  ]
```

To:
```
  "contactName": "RELATEDPARTY_1_NAME",
  "contactEmail": "RELATEDPARTY_1_EMAILADDRESS",
  "contactPhone": "RELATEDPARTY_1_PHONENUMBER",
  "contactAddress": {
    "address1": "RELATEDPARTY_1_STREET1",
    "name": "RELATEDPARTY_1_NAME",
    "city": "RELATEDPARTY_1_CITY",
    "zip": "RELATEDPARTY_1_POSTCODE",
    "country": "RELATEDPARTY_1_COUNTRY"
  }
```

Now only the first relation is rendered in the Mapper.

**Comment by madushajg** [comment]
> Does this also cover cases where the same top level array is used to derive values for several different top level values. This kind of structure is very common in my current work:
> 
> From:
> 
> ```
>   "relatedParty": [
>     {
>       "name": "RELATEDPARTY_1_NAME",
>       "role": "ORDER_CONTACT",
>       "contactMedium": [
>         {
>           "mediumType": "EMAIL",
>           "characteristic": {
>             "emailAddress": "RELATEDPARTY_1_EMAILADDRESS"
>           }
>         },
>         {
>           "mediumType": "TELEPHONE",
>           "characteristic": {
>             "phoneNumber": "RELATEDPARTY_1_PHONENUMBER"
>           }
>         },
>         {
>           "mediumType": "POSTAL_ADDRESS",
>           "characteristic": {
>             "city": "RELATEDPARTY_1_CITY",
>             "country": "RELATEDPARTY_1_COUNTRY",
>             "postCode": "RELATEDPARTY_1_POSTCODE",
>             "street1": "RELATEDPARTY_1_STREET1",
>             "name": "RELATEDPARTY_1_NAME"
>           }
>         }
>       ]
>     }
>   ]
> ```
> 
> To:
> 
> ```
>   "contactName": "RELATEDPARTY_1_NAME",
>   "contactEmail": "RELATEDPARTY_1_EMAILADDRESS",
>   "contactPhone": "RELATEDPARTY_1_PHONENUMBER",
>   "contactAddress": {
>     "address1": "RELATEDPARTY_1_STREET1",
>     "name": "RELATEDPARTY_1_NAME",
>     "city": "RELATEDPARTY_1_CITY",
>     "zip": "RELATEDPARTY_1_POSTCODE",
>     "country": "RELATEDPARTY_1_COUNTRY"
>   }
> ```
> 
> Now only the first relation is rendered in the Mapper.

@janihur That seems not covered by this scenario.
However we will add that to our backlog since that seems to be a valid and common requirement.
In that case, the source would be like the following.
```ballerina
import ballerina/io;

function transform(Input input) returns Output => (
    from var relatedPartyItem in input.relatedParty
        select {
            contactName: relatedPartyItem.name,
            contactEmail: (from ContactMediumItem item in relatedPartyItem.contactMedium
                          where item.mediumType == "EMAIL"
                          select <string>item.characteristic.emailAddress)[0],
            contactPhone: (from ContactMediumItem item in relatedPartyItem.contactMedium
                          where item.mediumType == "TELEPHONE"
                          select <string>item.characteristic.phoneNumber)[0],
            contactAddress: (from ContactMediumItem item in relatedPartyItem.contactMedium
                          where item.mediumType == "POSTAL_ADDRESS"
                          select {
                                address1: <string>item.characteristic.street1,
                                name: <string>item.characteristic.name,
                                city: <string>item.characteristic.city,
                                zip: <string>item.characteristic.postCode,
                                country: <string>item.characteristic.country
                          })[0]
        }
)[0];

type Characteristic record {
    string emailAddress?;
    string phoneNumber?;
    string city?;
    string country?;
    string postCode?;
    string street1?;
    string name?;
};

type ContactAddress record {
    string address1;
    string name;
    string city;
    string zip;
    string country;
};

type ContactMediumItem record {
    string mediumType;
    Characteristic characteristic;
};

type RelatedPartyItem record {
    string name;
    string role;
    ContactMediumItem[] contactMedium;
};

type Input record {
    RelatedPartyItem[] relatedParty;
};

type Output record {
    string contactName;
    string contactEmail;
    string contactPhone;
    ContactAddress contactAddress;
};

public function main() returns error? {
    json inputJson = check io:fileReadJson("resources/input.json");
    Input input = check inputJson.cloneWithType();
    Output output = transform(input);

    io:println(output.toJsonString());
}
```

**Comment by kaje94** [comment]
We've decided to come up with a uniform UX for all similar array member access scenarios.
Therefore this fix will not be available with v3.3.6
